mvc/src/main/java/nextstep/mvc/controller/tobe/AnnotationHandlerMapping.java

```diff
@@ -22,9 +30,31 @@ public AnnotationHandlerMapping(Object... basePackage) {
 
     public void initialize() {
         log.info("Initialized AnnotationHandlerMapping!");
+        Reflections reflections = new Reflections("com.techcourse.controller");
+        Set<Class<?>> controllerClass = reflections.getTypesAnnotatedWith(Controller.class);
+        for (Class<?> aClass : controllerClass) {
+            try {
+                Object clazz = aClass.getDeclaredConstructor().newInstance();
+                Set<Method> methods = ReflectionUtils.getAllMethods(aClass, ReflectionUtils.withAnnotation(RequestMapping.class));
+                for (Method method : methods) {
+                    RequestMapping annotation = method.getAnnotation(RequestMapping.class);
+                    handlerExecutions.put(
+                            new HandlerKey(annotation.value(), annotation.method()[0]),
+                            new HandlerExecution(clazz, method)
+                    );
+                }
+            } catch (InstantiationException | InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+                e.printStackTrace();
+            }
+        }
     }
 
     public Object getHandler(HttpServletRequest request) {
-        return null;
+        HandlerKey key = new HandlerKey(
+                request.getRequestURI(),
+                RequestMethod.requestMethod(request.getMethod())
+        );
+
+        return handlerExecutions.get(key);
```

Comment:
```suggestion
        return handlerExecutions.get(
            new HandlerKey(request.getRequestURI(), RequestMethod.valueOf(request.getMethod())));
```

Enum의 static method인 valueOf을 사용하면 RequestMethod에 있는 requestMethod 메소드를 사용하지 않아도 될 것 같아요!

Reply:
오호 감사합니다! 기본 제공 메서드가 있었군요~!